src/components/views/right_panel/UserInfo.tsx

```diff
@@ -1024,7 +1024,7 @@ const PowerLevelEditor: React.FC<{
 
         const myUserId = cli.getUserId();
         const myPower = powerLevelEvent.getContent().users[myUserId];
-        if (myPower && parseInt(myPower) === powerLevel) {
+        if (myPower && parseInt(myPower) <= powerLevel && myUserId !== target) {
```

Comment:
This still isn't comparing the new power level to the original level. The original level would be something like `powerLevelEvent.getContent().users[target]`.

Also, if that check is in place, you shouldn't need to check `myUserId !== target`, since it's not possible to promote yourself.

Reply:
@robintown **myPower** is the original power level of the present user logged in while **powerLevel** is the new power level.
The check between myPower and the new powerLevel is to warn the Admin that he is promoting a new user to his current powerLevel. Then the check between the myUserId and the target is to ensure the warning does not pop up if he is changing his own power level from **Admin** to Custom Level _100_ but only pop up when he is changing the power level of a different user.

Reply:
Ah, you're right, not sure what I was thinking earlier